scripts/py_matter_idl/matter_idl/generators/kotlin/MatterClusters.jinja

```diff
@@ -0,0 +1,168 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package matter.devicecontroller.cluster.clusters
+
+import java.util.ArrayList
+{% set typeLookup = idl | createLookupContext(cluster) %}
+class {{cluster.name}}Cluster(private val endpointId: UShort) {
+  companion object {
+    const val CLUSTER_ID: UInt = {{cluster.code}}u
+  }
+{% for command in cluster.commands | sort(attribute='code') -%}
+{%- set callbackName = command | javaCommandCallbackName() -%}
+{%- if not command.is_timed_invoke %}
+  fun {{command.name | lowfirst_except_acronym}}(callback: {{callbackName}}Callback
+{%- if command.input_param -%}
+{%- for field in (cluster.structs | named(command.input_param)).fields -%}
+  , {{field.name | lowfirst_except_acronym}}: {{encode_value(cluster, field | asEncodable(typeLookup), 0)}}
+{%- endfor -%}
+{%- endif -%}
+  ) {
+    // Implementation needs to be added here
+  }
+{%- endif %}
+
+  fun {{command.name | lowfirst_except_acronym}}(callback: {{callbackName}}Callback
+{%- if command.input_param -%}
+{%- for field in (cluster.structs | named(command.input_param)).fields -%}
+  , {{field.name | lowfirst_except_acronym}}: {{encode_value(cluster, field | asEncodable(typeLookup), 0)}}
+{%- endfor -%}
+{%- endif -%}
+  , timedInvokeTimeoutMs: Int) {
+    // Implementation needs to be added here
+  }
+{% endfor %}
+{%- set already_handled_command = [] -%}
+{%- for command in cluster.commands | sort(attribute='code') -%}
+{%- if command | isCommandNotDefaultCallback() -%}
+{%- set callbackName = command | javaCommandCallbackName() -%}
+{%- if callbackName not in already_handled_command %}
+  interface {{callbackName}}Callback {
+    fun onSuccess(
+{%- for field in (cluster.structs | named(command.output_param)).fields -%}
+    {{field.name | lowfirst_except_acronym}}: {{encode_value(cluster, field | asEncodable(typeLookup), 0)}}{% if not loop.last %}, {% endif %}
+{%- endfor -%}
+    )
+    fun onError(error: Exception)
+  }
+{% if already_handled_command.append(callbackName) -%}
+{%- endif -%}
+{%- endif -%}
+{%- endif -%}
+{%- endfor %}
+{%- set already_handled_attribute = [] -%}
+{% for attribute in cluster.attributes | rejectattr('definition', 'is_field_global_name', typeLookup) %}
+{%- set encodable = attribute.definition | asEncodable(typeLookup) -%}
+{%- set interfaceName = attribute | javaAttributeCallbackName(typeLookup) -%}
+{%- if interfaceName not in already_handled_attribute %}
+  interface {{interfaceName}} {
+    fun onSuccess(value: {{encode_value(cluster, encodable, 0)}})
+    fun onError(ex: Exception)
```

Comment:
onError and onSubscriptionEstablished are always the same. Would it make sense to have a common base interface for them (so that a generic error handler could be written for anything) or would that break some ABI that already exists?

Same for common functions above (e.g. I see on error in other places too)